content/docs/glossary/claimable-balance.mdx

```diff
@@ -49,12 +49,57 @@ This operation will load the ClaimableBalanceEntry that corresponds to the Balan
 Once a ClaimableBalanceEntry has been claimed, it will be deleted.
 
 ## Example
-The below code uses the [Go SDK](../software-and-sdks/index.mdx) to demonstrate how an account ("Account A") can create a `ClaimableBalanceEntry` with two Claimants, Account A (itself) and "Account B", another recipient.
+The below code demonstrates via both the JavaScript and Go [SDKs](../software-and-sdks/index.mdx) how an account ("Account A") can create a `ClaimableBalanceEntry` with two Claimants: Account A (itself) and "Account B" (another recipient).
 
 Each of these accounts can only claim the balance under certain, individual conditions. Namely, Account B has a full minute to claim the balance, after which Account A can "reclaim" the balance back for itself.
 
+It's worth emphasizing that there is no "recovery" mechanism for a claimable balance in general: if none of the predicates can be fulfilled, the balance **cannot be recovered**. The "reclaim" paradigm below acts as a safety net for this situation.
+
 <CodeExample>
 
+```js
+const sdk = require('stellar-sdk');
+
+var server = new sdk.Server("https://horizon-testnet.stellar.org")
+
+var A = sdk.Keypair.fromSecret("SAQLZCQA6AYUXK6JSKVPJ2MZ5K5IIABJOEQIG4RVBHX4PG2KMRKWXCHJ")
+var B = sdk.Keypair.fromPublicKey("GAS4V4O2B7DW5T7IQRPEEVCRXMDZESKISR7DVIGKZQYYV3OSQ5SH5LVP")
+
+// NOTE: Error checks are omitted for brevity; always validate things!
+
+server
+  .loadAccount(A.publicKey())
+  .then(async function(aAccount) {
+    // Create a claimable balance with our two above-described conditions.
+    let soon = Math.ceil((Date.now() / 1000) + 60) // .now() is in ms
+    let bCanClaim = sdk.Claimant.predicateBeforeRelativeTime("60")
+    let aCanReclaim = sdk.Claimant.predicateNot(
+      sdk.Claimant.predicateBeforeAbsoluteTime(soon.toString())
+    )
+
+    // Create the operation and submit it in a transaction.
+    let claimableBalanceEntry = sdk.Operation.createClaimableBalance({
+      claimants: [
+        new sdk.Claimant(B.publicKey(), bCanClaim),
+        new sdk.Claimant(A.publicKey(), aCanReclaim)
+      ],
+      asset: sdk.Asset.native(),
+      amount: "420",
+    })
+
+    let tx = new sdk.TransactionBuilder(aAccount, {fee: sdk.BASE_FEE})
+      .addOperation(claimableBalanceEntry)
+      .setNetworkPassphrase(sdk.Networks.TESTNET)
+      .setTimeout(180)
+      .build()
+
+    tx.sign(A)
+    let txResponse = await server.submitTransaction(tx)
```

Comment:
Clearly a knowledge gap on my part, thanks!

In general, how should one go about doing things that need waiting-on (like `server.loadAccount`, for example) in a `.then()` body? Or are those supposed to be delegated to their own `.then()` link in the Promise chain? In other words, would it have been more idiomatic of me to do something like 
```js
  return server.submitTransaction(tx)
}).then(function(txResponse) {
  console.log("Claimable balance created!")
  // The rest...
})
```
here, instead?

Reply:
If you go with promises you will want to maintain the depth of your code to 1 to avoid [hadouken code](https://www.reddit.com/r/ProgrammerHumor/comments/27yykv/indent_hadouken/).

**Desired promise structure**:
```js
server.submitTx(tx)     // async function
.then((result1) => {
  // do some stuff with result1 here if needed
  return server.applyResult1(result1)
})
.then((finalResult) => {
  // do some stuff with finalResult here if needed
  return server.finalize(finalResult)
})
.catch((err) => {
  handleErr(err)
})
```
*Please note the code above has depth=1. The depth can remain 1 regardless of the number of chained promises.*

**Avoidable hadouken code**:
```js
server.submitTx(tx)     // async function
.then((result1) => {
  // do some stuff with result1 here if needed
  server.applyResult1(result1)
  .then((finalResult) => {
    // do some stuff with finalResult here if needed
    return server.finalize(finalResult)
  })
})
.catch((err) => {
  handleErr(err)
})
```
*Note the code above has depth=2. The depth will increase proportionally with the number of chained promises.*

--
Of it doesn't make such a difference if the depth is 2 but it could escalate to something undesirable.